lib/matplotlib/figure.py

```diff
@@ -1576,6 +1576,110 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
             # Returned axis array will be always 2-d, even if nrows=ncols=1.
             return axarr
 
+    def build_grid(self, layout, *, subplot_kw=None, gridspec_kw=None,
+                   empty_sentinel=None):
+        """Build a layout based on an grid of strings.
+
+        This is a helper function to build complex GridSpec layouts visually.
+
+        Parameters
+        ----------
+        layout : list of list of {str, int,  Axes} or nested
+
+            A visual layout of how you want your Axes to be arranged
+            labeled as strings.  For example ::
+
+               x = [['A', 'A', 'B'],
+                    ['C', 'D', 'B']]
+
+            Produces 4 axes:
+
+            - 'A' which is 1 row high and spans the first two columns
+            - 'B' which is 2 rows high and is on the right edge
+            - 'C' which in 1 row and 1 column wide in the bottom left
+            - 'D' which in 1 row and 1 column wide in the bottom center
+
+        subplot_kw : dict, optional
+            Dict with keywords passed to the
+            :meth:`~matplotlib.figure.Figure.add_subplot` call used to create
+            each subplot.
+
+        gridspec_kw : dict, optional
+            Dict with keywords passed to the
+            `~matplotlib.gridspec.GridSpec` constructor used to create
+            the grid the subplots are placed on.
+
+        empty_sentinel : object, optional
+            Entry in the layout to mean "leave this space empty"
```

Comment:
Do we really want to make this configurable?
I guess someone could want to have a dict key of None, but... heh.

Reply:
The thought here was to allow pure-int inputs rather than forcing mixed type input.

Reply:
sure